include/lsst/afw/math/FunctionLibrary.h

```diff
@@ -836,7 +838,7 @@ class Chebyshev1Function2 : public BasePolynomialFunction2<ReturnT> {
     /**
      * Get x,y range
      */
-    lsst::geom::Box2D getXYRange() const {
+    lsst::geom::Box2D getXYRange() const noexcept {
```

Comment:
Looks like the `Box2D` constructor used here is not `noexcept` (though I believe it could be).

In addition, the `Point2D` constructors used hare are only conditionally `noexcept`, so this should be too, if you fix the `Box` issue - but I think it'd be better to `static_assert` against anything that could make the `Point2D` constructors throw instead, as per my comments on the `geom` PR.

Reply:
`Point2D` is `noexcept` (`2 != Eigen::Dynamic`, and `double` is nothrow-copyable). In fact, it's impossible to speak of a class as being conditionally `noexcept`, because there's nothing to condition on.

As for `Box2D`, I left that constructor `noexcept` because it could potentially throw when `max > min` (which is prevented in this case by `Box2D`'s class invariant). Though I guess you could argue that such a change in behavior would break backward-compatibility in ways other than unexpectedly throwing an exception...

Reply:
Yeah, on second thought I was being too pedantic. I'll update the `Box` constructors to guarantee that `max > min` never throws; I think that will leave `Box2I` overflow as the only failure mode.